lib/liquid/block_body.rb

```diff
@@ -154,14 +154,22 @@ def render_to_output_buffer(context, output)
     private
 
     def render_node(context, output, node)
-      node.render_to_output_buffer(context, output)
+      return if node.is_a?(Tag) && node.disabled?(context, output)
+      disable_tags(context, node.is_a?(Tag) ? node.disabled_tags : nil) do
```

Comment:
Could we move all this logic in `Tag#render_to_output_buffer` so we don't have to check for `node.is_a?(Tag)` here?

Reply:
Sadly not as almost every tag overrides render_to_output and doesn't run super

Reply:
This is all because `Variable` isn't a tag but its treated like one. When I did Go implementation I ended up making Variable extend Tag (Identical to the new echo tag) so that all Nodes were Tags.

Reply:
`Variable` is nested pretty deep so I thought, leave it for another PR and then this can come out.

Reply:
I could possibly add these methods as blank on Variable and drop these.

Reply:
Could we instead define `disabled?` and `disabled_tags` on all nodes? (return` false` / `[]` by default)